misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJob.kt

```diff
@@ -55,6 +58,31 @@ internal class SqsJob(
     metrics.jobsDeadLettered.labels(queueName.value, queueName.value).inc()
   }
 
+   fun setVisibilityTimeout(maxRetryCount: Int = 10) {
+    val visibilityTime = calculateVisibilityTimeOut(maxRetryCount)
+    queue.call { client ->
+      client.changeMessageVisibility(
+        ChangeMessageVisibilityRequest()
+          .withQueueUrl(queue.url)
+          .withReceiptHandle(message.receiptHandle)
+          .withVisibilityTimeout(visibilityTime)
+      )
+    }
+    metrics.visibilityTime.labels(queueName.value, queueName.value).set(visibilityTime.toDouble())
+  }
+
+  private fun calculateVisibilityTimeOut(maxReceiveCount: Int): Int {
```

Comment:
Ideally we would want to fetch the maxReceiveCount from the queue itself but it would mean making a call to fetch queue attributes so for simplicity sake I just let the users specify this number.  

I'm open to explore fetching this attribute somewhere in the queue setup process and providing it to the ResolvedQueue.

Reply:
that value can change after the value is retrieved. I feel this approach is good enough for now. it's relatively easy for oncall check these values if they're not seeing the behavior they're expecting.